### PR TITLE
Reduce the amount of spam displayed

### LINKED ISSUE #183: Reduce the amount of spam displayed by the `git_machete.tests.functional.test_machete.MacheteTester.setUp` and `setup_discover_standard_tree`
There's a looot of spam coming from the sandbox setup code, as you can see in the CI logs:

image

Pls find out a way to reduce the verbosity... probably passing `--quiet`/`-q` flag to the `git` commands invoked by the test setup code could be the way?

### PR DESCRIPTION
**Pull request by garg-tanishka**
#183 Before the number of lines was - 930, Now they are 746

### CODE REVIEW

**Comment by PawelLipski** [comment]
Okay... the base branch for the PR is now `develop` indeed (you didn't need to close the old one, yan can just `Edit` in the top-left corner and change the base)... pls also rebase your commit onto `develop` branch!

**Comment by PawelLipski** [comment]
Or `git cherry-pick`, might be even smoother

**Comment by PawelLipski** [comment]
Hmmm AFAICS there's still quite a few logs from `git` in the output anyway...

image

How about taking the more systematic approach (as suggested in the issue) - reimplement the `execute` method so that the output is always captured and not printed to stdout? Consider using `os.popen` or sth similar. This way you won't need to worry about finding a proper flag to pass to each `git` command

**Comment by garg-tanishka** [comment]
Before, the number of lines of the log was 1178. I tried with os.popen which resulted in 928 lines. But with subprocess.check_output resulted in 938 lines.

**Comment by PawelLipski** [comment]
Oh okay... I see why there still quite a lot of spam in the output! It's no longer coming from `git_machete.tests.functional.test_machete.GitRepositorySandbox.execute`, but probably from `git_machete.tests.functional.test_machete.MacheteTester.launch_command` instead

**Comment by PawelLipski** [comment]
Could you trace the exact source of logs like:

image

and find out a remedy to switch off that output in the tests?

My suspicion is that `git_machete.git_operations.GitContext.run_git` prints output to the stdout even when running within the context of `contextlib.redirect_stdout` (via `MacheteTester.launch_command`)... pls research!

File: git_machete/tests/functional/test_machete.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def check_out(self, branch: str) -> "GitRepositorySandbox":
     def commit(self, message: str = "Some commit message.") -> "GitRepositorySandbox":
         f = "%s.txt" % "".join(random.choice(string.ascii_letters) for _ in range(20))
         self.execute(f"touch {f}")
-        self.execute(f"git add {f}")
+        self.execute(f"git add {f} >/dev/null")
```
Comment (PawelLipski) [inline git_machete/tests/functional/test_machete.py]: All these changes to the git commands themselves aren't probably needed now that you use `check_output`

**Comment by garg-tanishka** [comment]
Can we close this PR and merge? Cause the issue as stated has been resolved. And for this issue, we can raise a new PR?

**Comment by PawelLipski** [comment]
> Can we close this PR and merge? Cause the issue as stated has been resolved. And for this issue, we can raise a new PR?

Ok! let's do it this way, you can take a look at #207  then... approving this PR

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**